src/hackney_manager.erl

```diff
@@ -503,7 +503,7 @@ handle_exit(Pid, {Ref, stream}, Reason, State) ->
     [{Ref, {Owner, Pid, #request_info{pool=Pool}=Info}}] ->
       %% unlink the owner
       unlink(Owner),
-      Pids2 = dict:erase(Pid, Pids1),
+      Pids2 = dict:erase(Owner, Pids1),
```

Comment:
actually i'm wondering what happen when a owner listen on 2 streams. Maybe it shouldn't be removed? I will test it.

Reply:
Hmm, that is a good point.  Though if an owner can listen on 2 streams does that not make the whole idea of having a dict of owner-pid -> stream data a bit flawed?

Reply:
@obmarg I think that's why you have the owner and the stream. This is something I want to fix anyway. We should probably only manage streams and link them directly to their owners...